bundles/org.openhab.binding.linktap/src/main/java/org/openhab/binding/linktap/internal/PollingDeviceHandler.java

```diff
@@ -0,0 +1,324 @@
+    protected String registeredDeviceId = EMPTY_STRING;
+
+    private final Object pollLock = new Object();
+
+    protected ExpiringCache<String> lastPollResultCache = new ExpiringCache<>(Duration.ofSeconds(5),
+            PollingDeviceHandler::expireCacheContents);
+
+    private final Object schedulerLock = new Object();
+    private @Nullable ScheduledFuture<?> backgroundGwPollingScheduler;
+
+    private final Object readBackPollLock = new Object();
+    private @Nullable ScheduledFuture<?> readBackPollSf = null;
+
+    protected volatile LinkTapDeviceConfiguration config = new LinkTapDeviceConfiguration();
+
+    protected void requestReadbackPoll() {
+        synchronized (readBackPollLock) {
+            cancelReadbackPoll();
+            scheduler.schedule(() -> {
+                pollForUpdate(true);
+            }, 750, TimeUnit.MILLISECONDS);
+        }
+    }
+
+    protected void cancelReadbackPoll() {
+        synchronized (readBackPollLock) {
+            ScheduledFuture<?> readBackPollSfRef = readBackPollSf;
+            if (readBackPollSfRef != null) {
+                readBackPollSfRef.cancel(false);
+                readBackPollSf = null;
+            }
+        }
+    }
+
+    private volatile long lastStatusCommandRecvTs = 0L;
+
+    public PollingDeviceHandler(final Thing thing) {
+        super(thing);
+    }
+
+    private void startStatusPolling() {
+        synchronized (schedulerLock) {
+            cancelStatusPolling();
+            backgroundGwPollingScheduler = scheduler.scheduleWithFixedDelay(() -> {
+                if (lastStatusCommandRecvTs + 135000 > System.currentTimeMillis()) {
+                    return;
+                }
+                pollForUpdate(false);
+            }, 1, 10, TimeUnit.SECONDS);
+        }
+    }
+
+    private void cancelStatusPolling() {
+        synchronized (schedulerLock) {
+            final ScheduledFuture<?> ref = backgroundGwPollingScheduler;
+            if (ref != null) {
+                ref.cancel(true);
+                backgroundGwPollingScheduler = null;
+            }
+        }
+    }
+
+    private static @Nullable String expireCacheContents() {
+        return null;
+    }
+
+    @Override
+    public void initialize() {
+        updateStatus(ThingStatus.UNKNOWN);
+        config = getConfigAs(LinkTapDeviceConfiguration.class);
+        final LinkTapBridgeHandler bridge = (LinkTapBridgeHandler) getBridgeHandler();
+        if (bridge == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "Bridge is not selected / set");
+            return;
+        } else if (bridge.getThing().getStatus().equals(ThingStatus.OFFLINE)) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE);
+            return;
+        }
+
+        scheduler.execute(() -> {
+            if (bridge.getThing().getStatus() == ThingStatus.ONLINE) {
+                initAfterBridge(bridge);
+            }
```

Comment:
Prevent double case by using the java 17 pattern matching for instanceoff
```suggestion
        if (!(getBridgeHandler() instanceof LinkTapBridgeHandler bridgeHandler)) {
            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "Bridge is not selected / set");
            return;
        } else if (bridgeHandler.getThing().getStatus().equals(ThingStatus.OFFLINE)) {
            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE);
            return;
        }

        scheduler.execute(() -> {
            if (bridgeHandler.getThing().getStatus() == ThingStatus.ONLINE) {
                initAfterBridge(bridgeHandler);
```